src/rust/sysmon-subgraph-generator/src/metrics.rs

```diff
@@ -0,0 +1,25 @@
+use failure::Error;
+use grapl_observe::metric_reporter::{MetricReporter, TagPair};
+use log::*;
+
+#[derive(Clone)]
+pub struct SysmonSubgraphGeneratorMetrics {
```

Comment:
the idea being we just add a bunch of functions here to do the actual logging; and keep passing around the SysmonSubgraphGeneratorMetrics object